contracts/Depository.sol

```diff
@@ -208,10 +209,11 @@ contract Depository is IErrorsTokenomics {
             revert LowerThan(vesting, MIN_VESTING);
         }
 
-        // Check for the expiration time overflow
-        uint256 expiry = block.timestamp + vesting;
-        if (expiry > type(uint32).max) {
-            revert Overflow(expiry, type(uint32).max);
+        // Check for the maturity time overflow for the current timestamp considering that the product will be live
+        // for at least MIN_VESTING time
+        uint256 maturity = block.timestamp + MIN_VESTING + vesting;
+        if (maturity > type(uint32).max) {
+            revert Overflow(maturity, type(uint32).max);
```

Comment:
Check that the vesting does not overflow when converted to the maturity time once the bond is created.